src/dataframe/dataframe.jl

```diff
@@ -366,8 +361,7 @@ end
 @inline function Base.getindex(df::DataFrame, row_inds::AbstractVector{T},
                                col_inds::Union{AbstractVector, Regex, Not, Between, All}) where T
     @boundscheck if !checkindex(Bool, axes(df, 1), row_inds)
-        throw(BoundsError("attempt to access a data frame with $(nrow(df)) " *
-                          "rows at index $row_inds"))
+        throw(BoundsError(df, (row_inds, col_inds)))
```

Comment:
Something that is lost in the change is that the user doesn't know whether the error comes from rows or column indices. For arrays that's not really an issue since it's easy to check which of the indices is out of range. But for data frames we can't print easily whether specified columns exist or not.

One possible solution would be to define our custom method for `showerror(::IO, BoundsError{<:AbstractDataFrame})`. Then we'd pass `(row_inds, :)` here, and `showerror` would print a message similar to the current one, mentioning a problem with row indices.

Reply:
This is an excellent point. I will rework the PR in full to follow this suggestion.